dvc/output/base.py

```diff
@@ -470,6 +471,48 @@ def move(self, out):
         if self.scheme == "local" and self.use_scm_ignore:
             self.repo.scm.ignore(self.fspath)
 
+    def transfer(
+        self,
+        source,
+        jobs=None,
+        update=False,
+        remote=None,
+        command=None,
+        no_progress_bar=False,
+    ):
+        from dvc.fs import get_cloud_fs
+
+        remote = self.repo.cloud.get_remote(remote, command)
+        odb = remote.odb
```

Comment:
For `to-cache` it will be `self.odb`, right? Then maybe consider passing remote `odb` to `out.transfer` so it could use it and if not specified - fallback to self.odb.